pkg/controller/vsphere/reconciler_test.go

```diff
@@ -95,130 +96,234 @@ func TestClone(t *testing.T) {
 		},
 	}
 
+	// Set this value to 2 because it's the default number on machine in mocked environment
+	vmsCount := 2
+
 	testCases := []struct {
-		testCase    string
-		machine     func(t *testing.T) *machinev1.Machine
-		expectError bool
-		cloneVM     bool
+		testCase              string
+		cloneVM               bool
+		expectedError         error
+		setupFailureCondition func()
```

Comment:
Would be better to return an error from this func and then `t.Fatal`, at the moment you are calling t.Fatal on the whole test `t`, not on the subtest `t`